bin/eds-import-from-figma.js

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env node
+(async function () {
+  const fs = require('fs');
+  const jsonfile = require('jsonfile');
+  const yargs = require('yargs/yargs');
+
+  // eslint-disable-next-line import/extensions
+  const { hideBin } = require('yargs/helpers');
+  const { prompt } = require('enquirer');
+  const at = require('lodash/at');
+  const set = require('lodash/set');
+
+  const { default: ora } = await import('ora');
+  const chalk = require('chalk');
+
+  // Set up the command structure to output help if the file to load is not specified
+  const args = yargs(hideBin(process.argv))
+    .command('$0 <file>', 'Figma variable file to import from', (yargs) => {
+      yargs.positional('file', {
+        describe: 'JSON tokens file emitted from the plugin',
+        type: 'string',
+      });
+    })
+    .option('verbose', {
+      describe: 'Print additional details for debugging purposes',
+      type: 'boolean',
+    })
+    .option('dry-run', {
+      describe: 'Run the script without writing any changes',
+      type: 'boolean',
+    }).argv;
+
+  // read in the config from config file, package json "eds", etc.
+  const { getConfig } = require('./_util');
+  const config = await getConfig();
+
+  const isVerbose = args.v;
+  const canWrite = !args.dryRun;
+
+  // Read the figma import file, and parse out the modes to select from, prompting the user to pick one
+  let importTheme;
+  try {
+    importTheme = JSON.parse(fs.readFileSync(args.file, 'utf8'));
+  } catch (e) {
+    console.error(chalk.red('cannot load import file', args.file));
+    process.exit(1);
+  }
+
+  // now, load in the local theme file. We want to look up keys in there
+  let localTheme;
+  try {
+    localTheme = JSON.parse(
+      fs.readFileSync(`${config.src}app-theme.json`, 'utf8'),
+    );
+  } catch (e) {
+    console.error(
+      chalk.red('cannot load local theme file', `${config.src}app-theme.json`),
+    );
+    process.exit(2);
```

Comment:
Q for learning purposes: why does the 1st try-catch `process.exit(1)` but the 2nd try-catch `process.exit(2)`. What's the difference between the two

Reply:
It's mostly a UNIX-y thing. I was going to add in a variable where I would apply [`<<=`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Left_shift_assignment), so that there are "meaningful" exit codes that could be documented or used elsewhere.

Reply:
Yeah, it's a way to distinguish between different types of errors. Maybe not strictly needed, but I'll never say no to Unix-y things